rust/src/node_manager/lock_cache/shuttle_tests.rs

```diff
@@ -48,3 +54,101 @@ fn shuttle__cached_node_manager_multiple_get_on_same_id_insert_in_cache_only_onc
         100,
     );
 }
+
+/// Tests all permutations of operations and a small set of node IDs on the lock cache using
+/// shuttle. The test is repeated for different cache sizes to stress cache contention scenarios.
+/// The idea is to find cases where the lock cache guarantees are violated, i.e. an operation
+/// returns a reference to a non-existing node or an unexpected error occurs.
+/// The permutation generates known invalid operation sequences that are expected to fail, and
+/// these are reported but do not cause the test to fail.
+/// To facilitate debugging, each test case is serialized to a file before being executed, and
+/// can be reloaded by setting the `CARMEN_SHUTTLE_SERIALIZED_CASE` environment variable.
+#[test]
+fn shuttle__operation_permutations() {
+    const PERMUTATION_SHUTTLE_ITERATIONS: usize = 100;
+    if cfg!(not(feature = "shuttle")) {
+        return;
+    }
+    let current_dir = env::current_dir().unwrap();
+    let CONCURRENT_OPS: usize = 6; // Must be >= 2
```

Comment:
Do we really need 6? More test cases are nice, but if we could make this so fast that we can run it on every PR it would also be nice.
What we need is that the same operation can be executed at least twice alongside another operation, so maybe 3 is already sufficient, which should be much faster.